pkgs/by-name/xc/xcursor-pro/package.nix

```diff
@@ -0,0 +1,48 @@
+{
+  stdenvNoCC,
+  lib,
+  fetchurl,
+  variants ? [
+    "Dark"
+    "Light"
+    "Red"
+  ],
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "xcursor-pro";
+  version = "2.0.2";
+  srcs = map (
+    variant:
+    fetchurl {
+      url = "https://github.com/ful1e5/XCursor-pro/releases/download/v${version}/XCursor-Pro-${variant}.tar.xz";
+      hash =
+        {
+          Dark = "sha256-lqA7rt1+yqLRv99XccAcYhEMiFmmlgmmL1Sr57lYJ4s=";
+          Light = "sha256-irYZLu8E1VttcUXQ+7XfUKxKyT1MbesG/b03qRBubac=";
+          Red = "sha256-NWkWHTMbC6/AD5P19d65F3sTqxD+5ZLmtsN/5ljFMAM=";
+        }
+        .${variant};
+    }
+  ) variants;
+
+  sourceRoot = ".";
```

Comment:
Does this line actually do anything?

Reply:
`sourceRoot` is required when `srcs` is set. This package can have up to 3 different sources, which is parameterized as the `variants` argument.

Reply:
Really? I'm pretty sure I made a package with `srcs` without setting `sourceRoot`, and nothing bad happened.


https://github.com/NixOS/nixpkgs/pull/311463/files#diff-9142932a559ca73958acafa0a6aac853e9347800d10bf95db3e63b9c8e4b6434

Reply:
This is probably why. Your package likely only produced a single directory in `unpackPhase`, thus `sourceRoot` was not required.

https://github.com/NixOS/nixpkgs/blob/fb3bf1dec8573d3dd17fab205ecb88fddadc53c0/doc/stdenv/stdenv.chapter.md?plain=1#L580

Reply:
Makes sense. Feel free to resolve this particular thread.